tests/subsys/fs/nvs/src/main.c

```diff
@@ -484,6 +484,45 @@ void test_nvs_corrupted_sector_close_operation(void)
 	execute_long_pattern_write(max_id);
 }
 
+/**
+ * @brief Test case when storage become full, so only deletion is possible.
+ */
+void test_nvs_full_sector(void)
+{
+	int err;
+	ssize_t len;
+
+	fs.sector_count = 3;
+
+	err = nvs_init(&fs, DT_FLASH_DEV_NAME);
+	zassert_true(err == 0,  "nvs_init call failure: %d", err);
+
+	u16_t filling_id = 0;
+
+	while (1) {
+		len = nvs_write(&fs, filling_id, &filling_id,
+				sizeof(filling_id));
+		if (len == -ENOSPC) {
+			filling_id--;
+			break;
+		}
+		zassert_true(len == sizeof(filling_id), "nvs_write failed: %d",
+			     len);
+		filling_id++;
+	}
+
+	/* check whether can delete whatever from full storage */
+	err = nvs_delete(&fs, 1);
+	zassert_true(err == 0,  "nvs_delete call failure: %d", err);
+
+	/* the last sector is full now, test re-initialization */
+	err = nvs_init(&fs, DT_FLASH_DEV_NAME);
+	zassert_true(err == 0,  "nvs_init call failure: %d", err);
+
+	len = nvs_write(&fs, 1, &filling_id, sizeof(filling_id));
+	zassert_true(len == sizeof(filling_id), "nvs_write failed: %d", len);
+}
+
```

Comment:
Consider adding a test to check if delete ATE can be written to the NVS with an ID that doesn't match any other IDs.

Reply:
good point

Reply:
test of delete behavior added